python/mrtrix3/commands/5ttgen/msmt.py

```diff
@@ -13,92 +13,116 @@
 #
 # For more details, see http://www.mrtrix.org/.
 
-import os
-import subprocess
-from mrtrix3 import MRtrixError, app, path, run, image
+from mrtrix3 import app, image, run
 
 
 def usage(base_parser, subparsers):  # pylint: disable=unused-variable
   parser = subparsers.add_parser('msmt', parents=[base_parser])
   parser.set_author('Arkiev D\'Souza (arkiev.dsouza@sydney.edu.au) & Robert E. Smith (robert.smith@florey.edu.au)')
-  parser.set_synopsis('Generate a 5TT image from ODF images.')
-
-  parser.add_argument('odf_wm', type=app.Parser.ImageIn(), help='The input white-matter ODF')
-  parser.add_argument('odf_gm', type=app.Parser.ImageIn(), help='The input grey-matter ODF')
+  parser.set_synopsis('Generate a 5TT image from ODF images')
+  parser.add_description('If the user does not manually provide a brain mask using the -mask option,'
+                         ' the command will automatically determine a mask for the output 5TT image;'
+                         ' this is necessary to conform to the expectations of the 5TT format.'
+                         ' This mask will be computed based on where the sum of the ODFs exceeds 50%'
+                         ' of what is expected from a voxel that contains a DWI signal'
+                         ' that exactly matches one of the response functions.')
+
+  parser.add_argument('odf_wm', type=app.Parser.ImageIn(), help='The input white matter ODF')
+  parser.add_argument('odf_gm', type=app.Parser.ImageIn(), help='The input grey matter ODF')
   parser.add_argument('odf_csf', type=app.Parser.ImageIn(), help='The input cerebrospinal fluid ODF')
   parser.add_argument('output', type=app.Parser.ImageOut(), help='The output 5TT image')
 
-  options = parser.add_argument_group('Options specific to the 'msmt' algorithm')
-  parser.add_argument('-mask', nargs='?', default=None, type=app.Parser.ImageIn(), help='The input binary brain mask image')
```

Comment:
@arkiev "`nargs='?'`" would make it inconsistent with the *MRtrix3* interface. The expectation is that the user either provides nothing, or the specify "`-mask`" followed by an image path. However "`nargs='?'`" specifically permits the absence of a trailing argument, which wouldn't make sense here.